contracts/V1/StateChannelDiamondProxy/DisputeManagerFacet.sol

```diff
@@ -99,121 +99,168 @@ contract DisputeManagerFacet is StateChannelCommon {
                 .executeStateTransitionOnState(channelId, encodedState, _tx);
     }
 
-    function getDispute(
-        bytes32 channelId
-    ) public view returns (Dispute memory) {
-        return disputes[channelId];
+    
+
+    function _handleBlockFraudProofs(
+       
+    ) internal {
+      
+       
+
+    }
+
+    function _handleDisputeFraudProofs(
+        // core specific checks
+        // apply proofs
+        // apply dispute to latest state
+        // set state
+    ) internal {
+
+    }
+
+    function _handleTimeoutDispute(
+
+    ) internal {
+
     }
 
     function createDispute(
         bytes32 channelId,
         uint forkCnt,
-        bytes memory encodedLatestFinalizedState,
-        bytes memory encodedLatestCorrectState,
-        ConfirmedBlock[] memory virtualVotingBlocks,
-        address timedoutParticipant,
-        uint foldedTransactionCnt,
-        Proof[] memory proofs
-    ) public {
-        require(!isDisputeInProgress(channelId), ErrorDisputeInProgrees());
-        require(getForkCnt(channelId) == forkCnt, ErrorDisputeForkMismatch());
+        bytes calldata encodedLatestState,
+        StateProof[] calldata stateProofs,
+        address[] calldata onChainSlashes,
+        Proof[] calldata proofs
+    ) public { 
+        // sanity checks
+        if(onChainSlashes.length != onChainSlashedParticipants.length 
+        || onChainSlashes[onChainSlashes.length - 1] != onChainSlashedParticipants[onChainSlashedParticipants.length - 1]) {
+            revert("onChainSlashes does not match onChainSlashedParticipants");
+        }
+        if(proofs.length == 0) return;
```

Comment:
also not true - it's possible to just timeout or self remove, but you shouldn't check this as a safety check either way, since we have a fraud proof for this, so if someone commits to nonsense they will get slashed - on-chain we just check the safety guards that can be affected by race condition on-chain (this is not affected by that)

Reply:
this also its a check per doc